admin-dev/themes/new-theme/js/pages/product/edit/product-options-manager.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import FormFieldToggler, {ToggleType} from '@components/form/form-field-toggler';
+import ProductSuppliersCollection from '@pages/product/components/suppliers/product-suppliers-collection';
+import {ProductSupplier, Supplier} from '@pages/product/components/suppliers/supplier-types';
+import SuppliersSelector from '@pages/product/components/suppliers/suppliers-selector';
+import ProductFormModel from '@pages/product/edit/product-form-model';
+import ProductMap from '@pages/product/product-map';
+import ProductConst from '@pages/product/constants';
+
+/**
+ * Manages product Options tab related components
+ */
+export default class ProductOptionsManager {
+  private readonly productType: string;
```

Comment:
Do we know every product type? I would list them in a enum so it's way stronger and use it as a type there

Reply:
HI @NeOMakinG 

we do have a restricted list of types for product. Although we still lack feedback from the community to know if it's too restrictive. But anyway for now we have only four so we could integrate the enum for now and change this solution later if it turns out it was a bad idea.

But maybe not in this PR though, it's not exactly the scope of this PR.

Reply:
Please keep it in your mind then, so we don't stack mistakes

If we type this as an enum, we will get stronger types avoiding bugs/mistakes in the future